pkg/cloudprovider/aws/instancetype.go

```diff
@@ -56,7 +56,7 @@ func (i *InstanceType) Architecture() string {
 			return value
 		}
 	}
-	return fmt.Sprint(aws.StringValueSlice(i.ProcessorInfo.SupportedArchitectures)) // Unrecognized, but used for error printing
+	return aws.StringValueSlice(i.ProcessorInfo.SupportedArchitectures)[0] // Unrecognized, return the first architecture value for error printing
```

Comment:
Not sure I understand why this was done?

Reply:
Previously this function returns the string representation of an array that contains the architecture values if those values cannot be translate. It was meant for outputting error messages.

Instead of outputting an array, now the function returns the first architecture value in that case. This will be a valid AWS specify architecture value. It won't change anything because the provisioner architecture requirement does not overlap with this AWS specify one. It is automatically ignored.